components/object_store/src/disk_cache.rs

```diff
@@ -124,13 +124,18 @@ struct DiskCache {
 }
 
 impl DiskCache {
-    fn new(root_dir: String, cap: usize, partition_bits: usize) -> Self {
-        let init_lru = || LruCache::new(cap);
-        Self {
+    fn try_new(root_dir: String, cap: usize, partition_bits: usize) -> Result<Self> {
+        let init_lru = |partition_num: usize| -> Result<_> {
+            let cap_per_par = cap / partition_num;
+            ensure!(cap_per_par != 0, InvalidCapacity);
+            Ok(LruCache::new(cap / partition_num))
+        };
+
+        Ok(Self {
             root_dir,
-            cap,
-            cache: PartitionedMutexAsync::new(init_lru, partition_bits, SeaHasherBuilder {}),
-        }
+            cap: cap / (1 << partition_bits),
```

Comment:
This is different with before, isn't just `cap` enough?

Reply:
This value is the capacity that each partition's LRU can hold and will be used for LRU.